### PR TITLE
Restrict hook order creation

### PR DESCRIPTION
**Pull request by danac-gs**
# Description
A merchant reported that 3 customer's orders are being duplicated many times in Magento. All orders have "Void" webhook failures associated with them

Fixes: https://app.asana.com/0/1125081140214268/1153726992405571/f
https://app.asana.com/0/941920570700290/1155745644813036/f

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

#changelog Restrict order creation from hook to pending and payment hooks only

# Type of change

- [x] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: Test/Unit/Helper/OrderTest.php [hunk 1]
```diff
@@ -822,4 +823,41 @@ public function additionAmountTotalProvider() {
...
+                )
+            );
+        }
+        $this->assertNull($this->currentMock->verifyOrderCreationHookType($hookType));
+    }
+
+    public function verifyOrderCreationHookTypeProvider()
+    {
+        return [
+            [true, OrderHelper::HT_PENDING],
```
[the remark this answers is not in the document]
Comment (danac-gs) [reply Test/Unit/Helper/OrderTest.php]: Most of the change requests were quite the opposite, in favor of using the constants. So I am leaving them, at least for now.

File: Helper/Order.php [hunk 2]
```diff
@@ -91,6 +91,11 @@ class Order extends AbstractHelper
     const TT_APM_PAYMENT = 'apm_payment';
     const TT_APM_REFUND = 'apm_refund';
 
+    // hook types
```
[the remark this answers is not in the document]
Comment (danac-gs) [reply Helper/Order.php]: Fair point. Thanks. Done.

File: Test/Unit/Helper/OrderTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (oxigenka) [reply Test/Unit/Helper/OrderTest.php]: Thanks, it was just for clarifying our code agreements regarding unit tests

**Comment by danac-gs** [comment]
@matt-thomason @njfaries are you familiar with the above error?

**Comment by njfaries** [comment]
@danac-gs looks like a broken peril rule. We'll get it fixed internally but don't worry about it here for now.

**Pull request merged**